packages/kit/src/runtime/app/forms.js

```diff
@@ -28,7 +28,13 @@ export function enhance(form, submit = () => {}) {
 			await invalidateAll();
 		}
 
-		if (location.origin + location.pathname === action.origin + action.pathname) {
+		// Don't apply action in case of success or invalid,
+		// because that updates the form prop which we don't want
```

Comment:
```suggestion
		// For success/invalid results, only apply action if it belongs to the,
		// current page, otherwise `form` will be updated erroneously
```